src/UI/Implementation/Component/Input/Field/Radio.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Input\Field;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Implementation\Component\Input\PostData;
+use ILIAS\UI\Implementation\Component\Input\NameSource;
+use ILIAS\UI\Implementation\Component\JavaScriptBindable;
+use ILIAS\UI\Implementation\Component\Triggerer;
+
+/**
+ * This implements the radio input.
+ */
+class Radio extends Input implements C\Input\Field\Radio, C\JavaScriptBindable{
+
+	use JavaScriptBindable;
+	use Triggerer;
+
+	/**
+	 * @var array <string,string> {$value => $label}
+	 */
+	protected $options = [];
+
+	/**
+	 * @var array <string,array> {$option_value => $fields}
+	 */
+	protected $dependant_fields = [];
+
+	/**
+	 * @inheritdoc
+	 */
+	protected function isClientSideValueOk($value) {
+		return true;
```

Comment:
Hmm, can we no do better here? Not sure however, give it a thougth if it's possible to restrict that somewhat more.